File: C++/BOJ/11723.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <string>
using namespace std;

int main()
{
    cin.tie(0);
    ios::sync_with_stdio(0);
    int m;
    cin>>m;
    vector<pair<int, int>> lst;
    for (int i = 1; i <= 20; i++)
        lst.push_back(pair(i, 0));
    for (int i = 0; i < m; i++)
    {
        string read;
        int num;
        cin>>read;
        if (read == "all")
        {
            for (int j = 0; j <= 20; j++)
                lst[j].second = 1;
        }
        else if (read == "empty")
        {
            for (int j = 0; j <= 20; j++)
                lst[j].second = 0;
        }
        else if (read == "add")
        {
            cin>>num;
            lst[num].second = 1;
        }
        else if (read == "check")
        {
            cin>>num;
            if (lst[num].second == 0)
                cout << 0 <<"\n";
            else
                cout << 1 << "\n";
        }
        else if (read == "remove")
        {
            cin>>num;
            lst[num].second = 0;
        }
        else if (read == "toggle")
        {
            cin>>num;
            if (lst[num].second == 0)
                lst[num].second = 1;
            else
                lst[num].second = 0;
        }
    }
}

File: Python/BOJ/11723.py
import sys

def solution():
    m = int(sys.stdin.readline().strip())
    dic = {n : 0 for n in range(1, 21)}
    for i in range(m):
        inp = sys.stdin.readline().strip()
        if inp == "all":
            dic = {n: 1 for n in range(1,21)}
        elif inp == "empty":
            dic = {n: 0 for n in range(1, 21)}
        else:
            command, num = inp.split()
            if command == "add":
                dic[int(num)] += 1
            elif command == "check":
                if dic[int(num)] == 0:
                    print(0)
                else:
                    print(1)
            elif command == "remove":
                dic[int(num)] = 0
            elif command == "toggle":
                if dic[int(num)] == 0:
                    dic[int(num)] += 1
                else:
                    dic[int(num)] = 0

solution()